service/dap/daptest/client.go

```diff
@@ -137,12 +137,16 @@ func (c *Client) InitializeRequest() {
 }
 
 // LaunchRequest sends a 'launch' request.
-func (c *Client) LaunchRequest(program string, stopOnEntry bool) {
+// Takes arguments as interface{} to allow for testing of values
+// of unexpected types since the launch flags are implementation-specific
```

Comment:
This seems a bit concerning from a typing / usability point of view. Functions that take long lists of arguments are not great in the first place, but when all these arguments are untyped it's even worse. It becomes pretty hard to use the function correctly.

What is the actual goal here? Are you testing that integers instead of string paths produce expected errors? Since we don't have Go on the other side, the mapping between types and what goes into JSON isn't very clear anyway. From a cursory look at the other changes in this PR it's really hard to notice where you're actually using non-standard types.

Would it make sense to have a simple-to-use `LaunchRequest` for the common cases (which will take part in many tests, I imagine), and some special version to test a few corner cases of type handling in isolated cases?

Reply:
I changed the code to use 2 versions of LaunchRequest. If you think I need different argument combinations, please let me know.

In general, LaunchRequest is the only struct where part of it won't be decoded with go types, but will be an untyped map. I added code in this CL to test not only for the presence of fields, but also for their type. And the new tests check that the code handles bad/missing values without panic.